src/__tests__/usePromiseState.ts

```diff
@@ -0,0 +1,125 @@
+import { renderHook } from '@testing-library/react-hooks';
+
+import * as hooks from '..';
+const { usePromise } = hooks;
+
+beforeEach(jest.clearAllMocks);
+
+it('starts with undefined promise.value', () => {
+  const {
+    result: {
+      current: [promise],
+    },
+  } = renderHook(usePromise);
+
+  expect(promise.value).toBeUndefined();
+});
+
+it('has promise.value after resolution', () => {
+  const result = 'foobar';
+  const {
+    result: {
+      current: [promise, resolve],
+    },
+  } = renderHook(usePromise);
+
+  resolve(result);
+
+  expect(promise.value).toBe(result);
+});
+
+it('resolves the promise', async () => {
+  const result = 'foobar';
+  const {
+    result: {
+      current: [promise, resolve],
+    },
+  } = renderHook(usePromise);
+
+  resolve(result);
+
+  expect(await promise).toBe(result);
```

Comment:
What do you think about using `await expect(promise).resolves.toBe(result);` instead?